pkg/discovery/discovery.go

```diff
@@ -80,6 +81,17 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) []error {
 		errlst = append(errlst, err)
 	}
 
+	// 8. Always write the done file
+	doneFile := cfg.ResultsDir + "/done"
```

Comment:
The only reason for `--no-exit` really is manual inspection of the container or manual copying of the results. I think you're right, this would feel much lighter coming from a script rather than the sonobuoy code. Let me revisit and see what I can do with that assumption.